common/src/types/v0/store/node.rs

```diff
@@ -1,95 +1,318 @@
 //! Definition of node types that can be saved to the persistent store.
-
-use crate::types::v0::{
-    openapi::models,
-    store::definitions::{ObjectKey, StorableObject, StorableObjectType},
-    transport::{self, NodeId},
+use crate::{
+    types::v0::{
+        openapi::models,
+        store::definitions::{ObjectKey, StorableObject, StorableObjectType},
+        transport::{self, NodeId},
+    },
+    IntoOption,
 };
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
+/// Generic labels associated with the node.
 pub type NodeLabels = HashMap<String, String>;
 
+/// Data relating to the cordoning of a node.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct CordonedState {
+    /// Labels used to enforce a node cordon.
+    pub cordonlabels: Vec<String>,
+}
+
+impl CordonedState {
+    /// Create a new CordonedState object.
+    pub fn new(cordonlabels: Vec<String>) -> Self {
+        Self { cordonlabels }
+    }
+    /// Add a cordon label to a CordonedState object.
+    pub fn add_cordon_label(&mut self, label: &String) {
```

Comment:
It's usually preferred to have `&str` instead of `&String` can be coerced into `&str`.

Reply:
fixed